pkg/trace/writer/sender.go

```diff
@@ -247,6 +250,14 @@ func (s *sender) sendPayload(p *payload) {
 			return
 		}
 		atomic.AddInt32(&s.attempt, 1)
+		atomic.AddInt32(&p.retries, 1)
+
+		if atomic.LoadInt32(&p.retries) > maxRetries {
+			//exceeded max retries per payload, dropping payload
```

Comment:
Same here and everywhere please 🙏 Also, this line is very obvious and does not require any additional comments for explaining.